app/src/main/java/org/simple/clinic/instantsearch/InstantSearchUpdate.kt

```diff
@@ -110,11 +114,24 @@ class InstantSearchUpdate @Inject constructor(
 
   private fun searchCriteriaFromInput(
       inputString: String,
-      additionalIdentifier: Identifier?
+      additionalIdentifier: Identifier?,
+      isInstantSearchByIdentifierEnabled: Boolean
   ): PatientSearchCriteria {
     return when {
-      digitsRegex.matches(inputString) -> PatientSearchCriteria.PhoneNumber(inputString.filterNot { it.isWhitespace() }, additionalIdentifier)
-      else -> PatientSearchCriteria.Name(inputString, additionalIdentifier)
+      digitsRegex.matches(inputString) -> numericPatientSearchCriteriaBasedOnFeatureFlag(isInstantSearchByIdentifierEnabled, inputString.filterNot { it.isWhitespace() }, additionalIdentifier)
+      else -> Name(inputString, additionalIdentifier)
+    }
+  }
+
+  private fun numericPatientSearchCriteriaBasedOnFeatureFlag(
```

Comment:
Not necessary to change this, but passing the feature flag as a method parameter here is unnecessary since the flag is already a property of `InstantSearchUpdate`.